src/main/scala/dpla/ingestion3/mappers/providers/Extractor.scala

```diff
@@ -9,4 +10,14 @@ import dpla.ingestion3.model.{DplaMapData, EdmAgent}
 trait Extractor {
   def build(rawData: String): DplaMapData
   def agent: EdmAgent
-}
+  /**
+    * Generates an MD5 checksum from the given providerId
+    * Implemented per provider
+    *
+    * DigestUtils.md5Hex(id)
+    *
+    * @return MD5 hash value
+    * @throws NotImplemented
+    */
+  def mintItemId(rawData: Any): String
```

Comment:
I think the best way to do it is you take the extraction of the needed value out of the responsibility of this method, and just make this method take the string and provider (see my comment below).